orderBook.js

```diff
@@ -0,0 +1,43 @@
+
+const reconcileOrder = (existingBook, incomingOrder) => {
+  let updatedBook = []
+  if (existingBook.length === 0 ||
+    existingBook[0].type === incomingOrder.type ||
+    existingBook[0].quantity != incomingOrder.quantity && existingBook[0].price != incomingOrder.price) {
+    updatedBook = existingBook.concat(incomingOrder)
+    return updatedBook
+  }
+  if (existingBook[0].quantity === incomingOrder.quantity) {
+    updatedBook = existingBook.filter(order => order.type != 'buy')
+    return updatedBook
+  }
+  if (existingBook[0].quantity > incomingOrder.quantity) {
+    existingBook[0].quantity = existingBook[0].quantity - incomingOrder.quantity
+    updatedBook = existingBook.reverse()
+    return updatedBook
+  }
+  if (existingBook[0].quantity < incomingOrder.quantity && existingBook.length < 3) {
```

Comment:
Not sure what you're trying to accomplish here with the length < 3 check but the length of the book has nothing to do with fulfillment beyond it being 0.